### PR TITLE
Add config flow tests

### PR DESCRIPTION
**Pull request by Bigdaddy1990**
## Summary
- add tests covering PawControl config flow and reauthentication
- ignore PytestRemovedIn9Warning for hass fixture

## Testing
- `pre-commit run --files tests/components/pawcontrol/conftest.py tests/components/pawcontrol/test_config_flow.py pyproject.toml`
- `pytest ./tests/components/pawcontrol --cov=custom_components.pawcontrol.config_flow --cov-report term-missing --cov-fail-under=0 --durations-min=1 --durations=0 --numprocesses=auto` *(fails: 'hass' fixture PytestRemovedIn9Warning)*

------
https://chatgpt.com/codex/tasks/task_e_68c74690b4888331b48ed30324af3716

___

### **PR Type**
Tests

___

### **Description**
- Add comprehensive config flow tests for PawControl integration

- Test user flow, duplicate dog ID validation, and reauthentication

- Configure pytest to ignore deprecated fixture warnings

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Test Setup"] --> B["Config Flow Tests"]
  B --> C["User Flow Test"]
  B --> D["Validation Test"]
  B --> E["Reauth Test"]
  F["Pytest Config"] --> G["Warning Filter"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>conftest.py</strong><dd><code>Add test configuration setup</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/components/pawcontrol/conftest.py

<ul><li>Add pytest configuration for Home Assistant custom component testing</ul>

</details>

  </td>
  <td><a href="https://github.com/Bigdaddy1990/pawcontrol/pull/439/files#diff-07fdf0be99aa7fd73f636fac3c1d3cfcce7ea4de1580e62dce157eb0d8e63bf5">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>test_config_flow.py</strong><dd><code>Add comprehensive config flow test suite</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tests/components/pawcontrol/test_config_flow.py

<ul><li>Add full user flow test covering pack creation and dog setup<br> <li> Add duplicate dog ID validation test<br> <li> Add reauthentication confirmation flow test<br> <li> Import required constants and test utilities</ul>

</details>

  </td>
  <td><a href="https://github.com/Bigdaddy1990/pawcontrol/pull/439/files#diff-858560014645d6b994337a7bb61acb6dae610fa646a970c80664a8c82bc939dd">+133/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyproject.toml</strong><dd><code>Suppress pytest fixture deprecation warning</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

pyproject.toml

- Add filter to ignore pytest deprecated 'hass' fixture warning

</details>

  </td>
  <td><a href="https://github.com/Bigdaddy1990/pawcontrol/pull/439/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: tests/components/pawcontrol/test_config_flow.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        result["flow_id"], user_input={"add_another": False}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "entity_profile"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"entity_profile": "standard"}
+    )
+    assert result["type"] == FlowResultType.CREATE_ENTRY
+    assert result["title"] == "My Pack (Standard)"
```
[the remark this answers is not in the document]
Comment (Bigdaddy1990) [reply tests/components/pawcontrol/test_config_flow.py:74]: @codex fix comments

**Comment by Bigdaddy1990** [comment]
@codex complete tests
